File: F#/0001multiples_of_3_or_5.fs

(*
.NET VERSION 7.0.410
0001multiples_of_3_or_5.fs
Copyright 2025 @GitHub JoergEm
This program is WITHOUT ANY WARRANTY; without even the implied
warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
*)

let summe =
    [1 .. 999]
    |> List.filter (fun nummer -> nummer % 3 = 0 || nummer % 5 = 0)
    |> List.sum
printfn "%d" summe

File: Spark/0001multiples_of3_or_5.py
"""
Python 3.8.20
PySpark 3.5.5
Pandas 2.0.3
0001multiples_of_3_or_5.py
Copyright 2025 @GitHub JoergEm
This program is WITHOUT ANY WARRANTY; without even the implied
warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
"""


from pyspark.sql import SparkSession
from pyspark.sql.functions import col


spark = SparkSession.builder.appName("0001multiples_of_3_or_5").getOrCreate()
numbers = spark.range(1000).toDF("nummer")
filtered = numbers.filter((col("nummer") % 3 == 0) | (col("nummer") % 5 == 0))
summe = filtered.groupBy().sum("nummer").collect()[0][0]
print(summe)
spark.stop()